connect_four.py

```diff
@@ -38,16 +37,15 @@ def check_col(self, col: str) -> ErrorCode:
         else:
             return ErrorCode.VALID
 
-    def __is_board_full(self) -> None:
+    def is_board_full(self) -> bool:
         if sum(len(col) for col in self.columns) >= self.num_cols * self.num_rows:
```

Comment:
You can make this more concise: 
return sum(len(col) for col in self.columns) >= self.num_cols * self.num_rows